src/Data/TypeRepMap/Internal.hs

```diff
@@ -464,11 +488,29 @@ fromSortedList l = runST $ do
                 writeArray result i (indexArray origin newFirst)
                 loop (2 * i + 2) (newFirst + 1)
 
+-- Returns a list of indexes which represents the "sorted" order of an array generated by
+-- fromSortedList of the provided length.
+-- I.e. fmap (fromSortedList [1, 2, 3, 4, 5] !!) (generateOrderMapping 5) == [1, 2, 3, 4, 5]
+generateOrderMapping :: Int -> [Int]
+generateOrderMapping len = runST $ do
+    orderMappingArr <- newPrimArray len
+    _ <- loop orderMappingArr 0 0
+    primArrayToList <$> unsafeFreezePrimArray orderMappingArr
+    where
```

Comment:
Just a minor stylistic note, but `where` is indented with two spaces according to our style guide.
```suggestion
  where
```